internal/provider/resource_tfe_workspace_agent_pool_execution_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// create and update
+// when workspace does not exist
+// when agent pool ID does not exist
+
+func TestAccTFEWorkspaceAgentPoolExecution_create_update(t *testing.T) {
+	tfeClient, err := getClientUsingEnv()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	org, orgCleanup := createBusinessOrganization(t, tfeClient)
+	t.Cleanup(orgCleanup)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEWorkspaceAgentPoolExecution_basic(org.Name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccTFECheckWorkspaceAgentPoolExists("tfe_agent_pool.pool"),
+					resource.TestCheckResourceAttr("tfe_agent_pool.pool", "organization_scoped", "false"),
+				),
+			},
+			{
+				Config: testAccTFEWorkspaceAgentPoolExecution_update(org.Name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccTFECheckWorkspaceAgentPoolExists("tfe_agent_pool.pool"),
+					resource.TestCheckResourceAttr("tfe_agent_pool.pool", "organization_scoped", "false"),
+				),
+			},
+			{
+				Config: testAccTFEWorkspaceAgentPoolExecution_destroy(org.Name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccTFECheckWorkspaceAgentPoolNotExists("tfe_agent_pool.pool"),
+				),
+			},
+		},
+	})
+}
+
+func testAccTFECheckWorkspaceAgentPoolExists(resource string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
```

Comment:
it seems like `testAccTFECheckWorkspaceAgentPoolExists` and `testAccTFECheckWorkspaceAgentPoolNotExists` are the same function, doing the same thing...

the check functions should be checking for the attachment/detachment as opposed to existence of agent pool.